firefox-ios/Client/Frontend/Fakespot/FakespotState.swift

```diff
@@ -176,4 +156,40 @@ struct FakespotState: ScreenState, Equatable {
 
         return state
     }
+
+    private static func handleTabDidReload(action: FakespotAction, state: FakespotState) -> FakespotState {
+        guard let tabUUID = action.tabUUID,
+                state.currentTabUUID == tabUUID,
+                let productId = action.productId
+        else { return state }
+
+        var state = state
+        state.telemetryState[tabUUID]?.adEvents[productId] = AdTelemetryState()
+        return state
+    }
+
+    private static func handlePressedShopping(state: FakespotState) -> FakespotState {
```

Comment:
Nit i would name the method `handlePressedShoppingButton` just to clear, and if methods grows it is not misleading the name

Reply:
Sure! I've made the change and rebased the branch.